### DISCUSSION TITLE
Export Keystone types for use by other packages

### DISCUSSION THREAD

**Discussion by xenobytezero** [keystonejs/keystone#7630] [Questions] [answered]
I was going to try and export some types from my Keystone packages that are based on my `Lists.<type>.Item` which are fully resolved, so I can use them in my client app. 

Is there any easy way for me to expose a few arbitrary types from the Keystone project? Can I hook into the build to do this?

**Comment by Achisingh** [comment, marked as answer]
Hi @xenobytezero in regards to 
> Is there any easy way for me to expose a few arbitrary types from the Keystone project? Can I hook into the build to do this?

There is nothing built-in to Keystone to help you do this. The easiest way is to copy it from Keystone to your client app. 

Although it is worth noting the `Lists.<type>.Item` type isn't what is returned from the GraphQL API, you may want to use a tool like https://www.graphql-code-generator.com/ to generate TypeScript types from GraphQL schema.

**Reply by xenobytezero** [reply]
Thanks for the reply.

This is essentially what I was doing already just wanted to check that I hadn't missed something.

Good recommendation of https://www.graphql-code-generator.com/, will take a look at that.
